buildtest/system.py

```diff
@@ -23,20 +24,55 @@ def __init__(self):
         """Constructor method for BuildTestSystem(). Defines all system configuration using
            class variable **system** which is a dictionary
         """
+        self.logger = logging.getLogger(__name__)
+        self.logger.debug("Starting System Compatibility Check")
         self.init_system()
-        self.system["SYSTEM"] = platform.system()
+        self.system["platform"] = platform.system()
         self.scheduler = self.check_scheduler()
         self.check_lmod()
 
+        if self.system["platform"] != "Linux":
+            print("System must be Linux")
+            sys.exit(1)
+
+        self.logger.debug("Finished System Compatibility Check")
+
     def init_system(self):
         """Based on the module "distro" set the linux distrubution name and version
         """
-        self.system["OS_NAME"] = distro.linux_distribution(
+
+        self.system["os"] = {}
+
+        self.system["os"]["name"] = distro.linux_distribution(
             full_distribution_name=False
         )[0]
-        self.system["OS_VERSION"] = distro.linux_distribution(
+        self.system["os"]["version"] = distro.linux_distribution(
             full_distribution_name=False
         )[1]
+        self.system["os"]["major_version"] = distro.major_version(best=True)
+        self.system["os"]["minor_version"] = distro.major_version(best=True)
+        self.system["os"]["build_number"] = distro.build_number(best=True)
+        self.system["env"] = dict(os.environ)
+        self.system["python"] = shutil.which("python") or None
+
+        self.logger.info(f"Operating System: {self.system['os']['name']}")
+        self.logger.info(f"Operating System Version: {self.system['os']['version']}")
+        self.logger.info(
+            f"Operating System Major Version: {self.system['os']['major_version']}"
+        )
+        self.logger.info(
+            f"Operating System Minor Version: {self.system['os']['minor_version']}"
+        )
+        self.logger.info(
+            f"Operating System Build Number: {self.system['os']['build_number']}"
+        )
+
+        self.logger.debug("Session Environment Variables")
+        self.logger.debug("{:_<80}".format(""))
+        for k, v in self.system["env"].items():
```

Comment:
This is actually a security risk - we should not be printing environment variables to a log file. If absolutely necessary, just print the keys.

Reply:
I would suggest namespacing here - for buildtest envars, we prefix with BUILDTEST and then we at least know the ones that are safe to skip / print, and we can filter os.environ down to anything that starts with `BUILDTEST` or another white listed set.

Reply:
Yea i used to have buildtest variables prefixed with BUILDTEST that's not the case, if you want i can just print the keys without the value.

Reply:
Ok let’s open an issue for the environment namespace, other than filtering we don’t want a conflict of envars with some other tool that happens to use a variable with the same name.

Reply:
okay for now i print the keys and its fixed in the logs.

Here is a snippet

```
2020-04-08 21:25:52,375 [system.py:70 - init_system() ] - [DEBUG] Session Environment Variables
2020-04-08 21:25:52,375 [system.py:71 - init_system() ] - [DEBUG] ________________________________________________________________________________
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] REMOTEHOST
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] CABAQUS
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] EBVERSIONZLIB
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] MANPATH
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] FIGS
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] EBVERSIONPYTHON
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] XDG_SESSION_ID
2020-04-08 21:25:52,375 [system.py:73 - init_system() ] - [DEBUG] CABAQUS_O

```